### PR TITLE
Python refactoring

### PR DESCRIPTION
**Pull request by Shadow-Devil**
Just some reformatting, type annotations, refactorings etc. of the python scripts to make the code more readable :)

### CODE REVIEW

File: t.py [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+import traceback
 import unittest
+from curses.ascii import isspace
+from multiprocessing.dummy import Pool
+from typing import Literal, Final
+
+import pyperclip
+from tqdm import tqdm
 
+# TODO: why rip here but not in a.py?
```
Comment (Shadow-Devil) [inline t.py:20]: Could you maybe explain if it was just forgotten or it's explicitly left out?

File: t.py [hunk 2]
```diff
@@ -79,7 +91,7 @@ class RegisterOperand(Operand):
 
     def __init__(self, name):
         self.name = name.lower()
-        self.size()
+        self.size()  # TODO: why call this here?
```
Comment (Shadow-Devil) [inline t.py:94]: I think this call is unnecessary, should I remove it?

File: t.py [hunk 3]
```diff
@@ -638,31 +668,32 @@ def permutate_with_flags(inputs: List[List[int]], flags_to_permutate: List[int])
...
 
     @staticmethod
-    def generate_inputs(dynamic_operands: List[Operand]) -> List[List[int]]:
+    def generate_inputs(dynamic_operands: list[Operand]) -> list[list[int]]:
         dynamic_operands = len(dynamic_operands)
 
         if dynamic_operands == 0:
             return [[]]
         elif dynamic_operands == 1:
+            # TODO: changed to [i] but was i
```
Comment (Shadow-Devil) [inline t.py:677]: I had a type error in here, so I changed it from `[i for i in range(0, 1024)]` to `[[i] for i in range(0, 1024)]` but I'm not sure if that breaks the code or the old version was correct.

File: t.py (hunk 1 above)
Comment (xarantolus) [reply t.py:20]: I think it was just forgotten in `a.py`, but not entirely sure. It's kind of rare to explicitly test rip-relative addressing in instruction tests (it's only done in a few `xor` tests), so I think it doesn't matter if it's there or not, we can leave it out if you want to

File: t.py (hunk 2 above)
Comment (xarantolus) [reply t.py:93]: The call is used for detecting invalid registers, so when passing a wrong name to the constructor the size() call will throw an exception

File: t.py (hunk 3 above)
Comment (xarantolus) [reply t.py:671]: Good spot! I think your version is correct, the previous one may have failed when generating the assembly file without me noticing (currently all exceptions there are ignored, as some invalid instructions will be generated and must be ignored, but it's not a great solution).

Comment (xarantolus) [reply t.py:671]: Actually it probably makes sense to only ignore one specific error type of assembling (`CalledProcessError`) in `learn_single_flags` and log all the others, making it instantly noticeable when a lot of things go wrong

**Comment by Shadow-Devil** [comment]
Hi @xarantolus I think this is now done :)
If you have any more remarks, please let me know.

**Comment by xarantolus** [comment]
Nice, that looks really great :) Thank you!

**Pull request merged**